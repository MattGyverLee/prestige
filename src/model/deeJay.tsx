import * as actions from "../store";

import React, { Component } from "react";
import { faLayerGroup, faVolumeUp } from "@fortawesome/free-solid-svg-icons";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import WaveSurfer from "wavesurfer.js";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { sourceAudio } from "./globalFunctions";

interface StateProps {
  timeline: any;
  sourceMedia: any;
  volumes: number[];
  waveSurfers: any;
  pos: number;
  playing: boolean;
}

interface DispatchProps {
  toggleWaveSurferPlay: typeof actions.toggleWaveSurferPlay;
  waveSurferPosChange: typeof actions.waveSurferPosChange;
  setWaveSurfer: typeof actions.setWaveSurfer;
  setWSVolume: typeof actions.setWSVolume;
}

interface DeeJayProps extends StateProps, DispatchProps {}

export class DeeJay extends Component<DeeJayProps> {
  componentDidMount = () => {
    [0, 1, 2].forEach((idx: number) => {
      this.props.setWaveSurfer(
        WaveSurfer.create({
          container: "#waveform" + idx.toString(),
          height: 50,
          barWidth: 1,
          cursorWidth: 1,
          backend: "MediaElement",
          progressColor: "#4a74a5",
          cursorColor: "#4a74a5",
          responsive: true,
          waveColor: "#ccc",
          hideScrollbar: true
        }),
        idx
      );
    });
  };

  testLoad = () => {
    if (this.props.waveSurfers[0] !== null) {
      this.props.waveSurfers[0].load(
        sourceAudio(this.props.sourceMedia)[0].blobURL
      );
      this.props.waveSurfers[0].on("waveform-ready", () =>
        this.onSurferReady(0)
      );
      this.props.setWSVolume(0, this.props.volumes[0]);
    }
  };

  handleTogglePlay() {
    this.props.toggleWaveSurferPlay();
  }

  handlePosChange = (e: any) => {
    this.props.waveSurferPosChange(e.originalArgs[0]);
  };

  onSurferReady = (idx: number) => {
    this.props.waveSurfers[idx].play();
  };

  setVolume = (e: any) => {
    this.props.setWSVolume(parseInt(e.target.id), e.target.value);
  };

  render() {
    const waveTableRows = [0, 1, 2].map((idx: number) => {
      return (
        <tr>
          <td className="wave-table-play">
            <FontAwesomeIcon icon={faVolumeUp} />
          </td>
          <td className="wave-table-overlay">
            <FontAwesomeIcon icon={faLayerGroup} />
          </td>
          <td className="wave-table-volume">
            <input
              id={idx.toString()}
              type="range"
              min={0}
              max={1}
              step={0.2}
              onChange={this.setVolume}
              value={this.props.volumes[idx]}
            />
          </td>
          <td className="waveform" id={"waveform" + idx.toString()}></td>
        </tr>
      );
    });

    return (
      <div>
        <button onClick={this.testLoad}></button>
        <div className="wave-table-container">
          <table className="wave-table">{waveTableRows}</table>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state: actions.StateProps): StateProps => ({
  pos: state.deeJay.pos,
  playing: state.deeJay.playing,
  timeline: state.annotations.timeline,
  waveSurfers: state.deeJay.waveSurfers,
  volumes: state.deeJay.volumes,
  sourceMedia: state.tree.sourceMedia
});

const mapDispatchToProps = (dispatch: any): DispatchProps => ({
  ...bindActionCreators(
    {
      waveSurferPosChange: actions.waveSurferPosChange,
      toggleWaveSurferPlay: actions.toggleWaveSurferPlay,
      setWaveSurfer: actions.setWaveSurfer,
      setWSVolume: actions.setWSVolume
    },
    dispatch
  )
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DeeJay);

/*        
<div className="mediaControls">
          <table>
            <tbody>
              <tr>
                <th>Volume</th>
                <td>
                  <input
                    max={1}
                    min={0}
                    onChange={this.onVolumeChange}
                    step="any"
                    type="range"
                    value={this.props.volume}
                  />
                </td>
              </tr>
              <tr>
                <th>
                  <label htmlFor="muted">Muted</label>
                </th>
                <td>
                  <input
                    checked={this.props.muted}
                    id="muted"
                    onChange={this.props.toggleMuted}
                    type="checkbox"
                  />
                </td>
              </tr>
              <tr>
                <th>Played</th>
                <td>
                  <progress max={1} value={this.props.played} />
                </td>
              </tr>
              <tr>
                <th>remaining</th>
                <td>
                  <Duration
                    className="Duration-Remaining"
                    seconds={(
                      this.props.duration *
                      (1 - this.props.played)
                    ).toFixed(3)}
                  />
                </td>
              </tr>
              <tr>
                <th>Loaded</th>
                <td>
                  <progress max={1} value={this.props.loaded} />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
*/
