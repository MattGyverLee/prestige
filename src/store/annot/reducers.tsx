import * as types from "./types";

export const annCleanStore: types.AnnotationState = {
  annotations: [],
  annotationSet: [],
  annotationTable: [
    {
      id: 0,
      startTime: 0,
      txtTransc: "Not Loaded",
    },
  ],
  audCarefulMain: false,
  audTranslMain: false,
  categories: [],
  fileInfoMain: false,
  sayMoreMetaMain: false,
  timeline: [],
  currentTimeline: -1,
  prevTimeline: -1,
  txtTranscMain: false,
  txtTranscSubtitle: false,
  txtTranslMain: false,
  txtTranslSubtitle: false,
  timelineChanged: false,
  timelinesInstantiated: false,
};

export function annotationReducer(
  state = annCleanStore,
  action: types.AnnotationActionTypes
): types.AnnotationState {
  switch (action.type) {
    // case types.HARD_RESET_APP: {
    //   return state;
    // }
    case types.ON_NEW_FOLDER: {
      return { ...state, timelineChanged: true };
    }
    case types.ON_RELOAD_FOLDER: {
      return { ...state, timelineChanged: true };
    }
    case types.SET_URL: {
      return {
        ...state,
        currentTimeline: action.payload.timelineIndex,
      };
    }
    case types.ADD_ORAL_ANNOTATION: {
      if (action.payload.idx === -1) {
        let eafFile = action.payload.newMilestone.data[0].data;
        eafFile = eafFile.substring(0, eafFile.indexOf("_Annotations"));
        const syncMedia = eafFile;
        eafFile = eafFile.substring(0, eafFile.lastIndexOf(".")) + ".eaf";
        const timeline: types.LooseObject = {
          milestones: [action.payload.newMilestone],
          eafFile: eafFile,
          syncMedia: [syncMedia],
          instantiated: true,
        };
        return { ...state, timeline: [...state.timeline, timeline] };
      }
      let added = false;
      let milestones = state.timeline[action.payload.idx].milestones.map(
        (m: types.LooseObject) => {
          if (
            m.startTime === action.payload.newMilestone.startTime &&
            m.stopTime === action.payload.newMilestone.stopTime
          ) {
            added = true;
            return {
              ...m,
              data: [...m.data, ...action.payload.newMilestone.data],
            };
          } else {
            return m;
          }
        }
      );
      const newM = action.payload.newMilestone;
      if (!added) {
        milestones = [...milestones, newM];
      }
      return {
        ...state,
        timeline: state.timeline.map((t: types.LooseObject, i: number) =>
          i === action.payload.idx ? { ...t, milestones } : t
        ),
      };
    }
    case types.ADD_CATEGORY: {
      return {
        ...state,
        categories: [...state.categories, action.payload],
      };
    }
    case types.PUSH_ANNOTATION: {
      return {
        ...state,
        annotations: [action.payload],
      };
    }
    case types.PUSH_ANNOTATION_TABLE: {
      return {
        ...state,
        annotationTable: action.payload,
        timelineChanged: false,
      };
    }
    case types.PUSH_TIMELINE: {
      return {
        ...state,
        timeline: state.timeline.concat(action.payload.timeline),
        timelineChanged: true,
      };
    }
    case types.TOGGLE_AUDCAREFUL_MAIN: {
      return {
        ...state,
        audCarefulMain: action.payload || !state.audCarefulMain,
      };
    }
    case types.TOGGLE_AUDTRANSL_MAIN: {
      return {
        ...state,
        audTranslMain: action.payload || !state.audTranslMain,
      };
    }
    case types.TOGGLE_AUDTRANSC_MAIN: {
      return {
        ...state,
        txtTranscMain: action.payload || !state.txtTranscMain,
      };
    }
    case types.TOGGLE_TRANSC_SUB: {
      return {
        ...state,
        txtTranscSubtitle: action.payload || !state.txtTranscSubtitle,
      };
    }
    case types.TOGGLE_TXTTRANSL_MAIN: {
      return {
        ...state,
        txtTranslMain: action.payload || !state.txtTranslMain,
      };
    }
    case types.TOGGLE_AUDTRANSL_SUB: {
      return {
        ...state,
        audTranslMain: action.payload || !state.audTranslMain,
      };
    }
    case types.TOGGLE_META_MAIN: {
      return {
        ...state,
        sayMoreMetaMain: action.payload || !state.sayMoreMetaMain,
      };
    }
    case types.TOGGLE_FILEINFO: {
      return {
        ...state,
        fileInfoMain: action.payload || !state.fileInfoMain,
      };
    }
    case types.FILE_DELETED: {
      const tempTimeline = state.timeline
        .map((t: types.LooseObject) => {
          const tempTimeline = {
            ...t,
            milestones: t.milestones
              .map((m: any) => {
                return {
                  ...m,
                  data: m.data.filter(
                    (md: any) =>
                      md.data !== action.payload &&
                      !(
                        t.eafFile === action.payload &&
                        !md.data.startsWith("file:///")
                      )
                  ),
                };
              })
              .filter((m: any) => m.data.length !== 0),
            syncMedia: t.syncMedia.filter((s: any) => s !== action.payload),
          };
          return tempTimeline;
        })
        .filter((t: any) => t.milestones.length !== 0);
      return {
        ...state,
        annotationTable: state.annotationTable.map(
          (annot: types.LooseObject) => {
            if (annot.audCareful === action.payload) {
              return { ...annot, audCareful: "" };
            } else if (annot.audTransl === action.payload) {
              return { ...annot, audTransl: "" };
            }
            return annot;
          }
        ),
        timeline: tempTimeline,
        timelineChanged: true,
      };
    }
    case types.UPDATE_PREV_TIMELINE: {
      return { ...state, prevTimeline: action.payload, timelineChanged: false };
    }
    case types.SET_TIMELINES_INSTANTIATED: {
      return { ...state, timelinesInstantiated: action.payload };
    }
    case types.SET_TIMELINE_CHANGED: {
      return { ...state, timelineChanged: action.payload };
    }
    case types.LOAD_ANNOT: {
      return { ...action.payload };
    }
    default:
      // console.log("Failed Tree Action", action);
      return state;
  }
}
