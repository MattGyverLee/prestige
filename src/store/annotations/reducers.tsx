// src/store/tree/reducers.ts
import * as annTy from "./types";
//import {map} from "redux-data-structures"

/* export function audCareful_Main() {
 return map({
  trueActionTypes: ['ENABLE_AUDCAREFUL_MAIN'],
  falseActionTypes: ['DISABLE_AUDCAREFUL_MAIN']
})
} */

const initialState: annTy.AnnotationState = {
  annotations: {},
  annotationSet: {},
  audCareful_Main: false,
  audTransl_Main: false,
  txtTransc_Main: false,
  txtTransc_Subtitle: false,
  txtTransl_Main: false,
  txtTransl_Subtitle: false,
  SayMoreMeta_Main: false,
  fileInfo_Main: false,
}

export function annotationReducer(
  state = initialState,
  action: annTy.AnnotationActionTypes
): annTy.AnnotationState {
  switch (action.type) {
    case annTy.RESET_ANNOTATION_SESSION: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.WIPE_ANNOTATION_SESSION: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.ADD_ANNOTATION: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.UPDATE_ANNOTATION: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.REMOVE_ANNOTATION: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.ADD_ANNOTATIONSET: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.UPDATE_ANNOTATIONSET: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.REMOVE_ANNOTATIONSET: {
      return {
        ...state,
        ...action.payload
      };
    }
    case annTy.ENABLE_AUDCAREFUL_MAIN: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_AUDCAREFUL_MAIN: {
      return {
        ...state
      };
    }
    case annTy.ENABLE_AUDTRANSL_MAIN: {
      return {
        ...state,
        audTransl_Main: true
      };
    }
    case annTy.DISABLE_AUDTRANSL_MAIN: {
      return {
        ...state,
        audTransl_Main: false
      };
    }
    case annTy.ENABLE_AUDTRANSC_MAIN: {
      return {
        ...state,
        txtTransc_Main: true
      };
    }
    case annTy.DISABLE_AUDTRANSC_MAIN: {

      return {
        ...state,
        txtTransc_Main: false
      };
    }
    case annTy.ENABLE_TRANSC_SUB: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_TRANSC_SUB: {
      return {
        ...state
      };
    }
    case annTy.ENABLE_TXTTRANSL_MAIN: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_TXTTRANSL_MAIN: {
      return {
        ...state
      };
    }
    case annTy.ENNABLE_AUDTRANSL_SUB: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_AUDTRANSL_SUB: {
      return {
        ...state
      };
    }
    case annTy.ENABLE_META_MAIN: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_META_MAIN: {
      return {
        ...state
      };
    }
    case annTy.ENABLE_FILEINFO: {
      return {
        ...state
      };
    }
    case annTy.DISABLE_FILEINFO: {
      return {
        ...state
      };
    }

    default:
      return state;
  }
}

