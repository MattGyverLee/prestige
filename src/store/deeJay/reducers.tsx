import * as types from "./types";

export const deeJayCleanStore: types.DeeJayState = {
  volumes: [1, 0, 0],
  dispatch: { dispatchType: "" },
  subtitle: "",
};

export function deeJayReducer(
  state = deeJayCleanStore,
  action: types.DeeJayActionTypes
): types.DeeJayState {
  switch (action.type) {
    case types.RESET_DEE_JAY: {
      state = deeJayCleanStore;
      return {
        ...state,
      };
    }
    case types.SET_WS_VOLUME: {
      return {
        ...state,
        volumes: state.volumes.map((v: number, idx: number) =>
          idx === action.payload.idx ? action.payload.volume : v
        ),
      };
    }
    case types.SET_DISPATCH: {
      return {
        ...state,
        dispatch: action.payload,
      };
    }
    case types.SET_SUBTITLE: {
      return {
        ...state,
        subtitle: action.payload,
      };
    }
    default: {
      return state;
    }
  }
}
