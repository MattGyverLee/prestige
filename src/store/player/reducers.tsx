import * as types from "./types";

export const speeds: number[] = [
  0.2,
  0.33,
  0.5,
  0.66,
  0.8,
  1,
  1.25,
  1.5,
  2,
  3,
  5,
];

export const playerCleanStore: types.MediaPlayerState = {
  controls: false,
  duration: 0,
  loaded: 0,
  loop: false,
  muted: true,
  playbackRate: 1.0,
  playbackMultiplier: 1.0,
  played: 0,
  playing: false,
  ready: false,
  seek: { time: -1, scale: "fraction" },
  seeking: false,
  speedsIndex: 5,
  url: "",
  volume: 0.8,
};

export function playerReducer(
  state = playerCleanStore,
  action: types.PlayerActionTypes
): types.MediaPlayerState {
  switch (action.type) {
    case types.HARD_RESET_APP: {
      state = playerCleanStore;
      return state;
    }
    case types.ON_NEW_FOLDER: {
      if (action.payload.blobURL !== undefined) {
        return {
          ...playerCleanStore,
          url: action.payload.blobURL,
        };
      }
      return playerCleanStore;
    }
    case types.UPDATE_PLAYER_SESSION: {
      // I might Deprecate This Action
      return {
        ...state,
        ...action.payload,
      };
    }
    case types.SET_URL: {
      return {
        ...playerCleanStore,
        url: action.payload.blobURL,
      };
    }
    case types.TOGGLE_PLAY: {
      return {
        ...state,
        playing: action.payload !== undefined ? action.payload : !state.playing,
      };
    }
    case types.STOP_PLAYING: {
      return {
        ...state,
        playing: false,
        url: "none",
      };
    }
    case types.TOGGLE_LOOP: {
      return {
        ...state,
        loop: !state.loop,
      };
    }
    case types.ON_PLAY: {
      return {
        ...state,
        // playing: true
        // todo: this seems circular
      };
    }
    case types.ON_PAUSE: {
      return {
        ...state,
        // playing: false,
        // todo: this seems circular
      };
    }
    case types.ON_ENDED: {
      return {
        ...state,
        playing: state.loop,
      };
    }
    case types.ON_READY: {
      return {
        ...state,
        ready: action.payload,
      };
    }
    case types.ON_PROGRESS: {
      if (!state.seeking && action.payload !== undefined) {
        return {
          ...state,
          ...action.payload,
        };
      } else {
        return state;
      }
    }
    case types.SET_DURATION: {
      return {
        ...state,
        duration: action.payload,
      };
    }
    case types.SET_PLAYBACK_RATE: {
      return {
        ...state,
        playbackRate:
          action.payload >= 15
            ? 14.5
            : action.payload <= 0.2
            ? 0.2
            : action.payload,
      };
    }
    case types.SET_PLAYBACK_MULTIPLIER: {
      return {
        ...state,
        playbackMultiplier: action.payload,
      };
    }
    case types.TOGGLE_MUTED: {
      return {
        ...state,
        muted: !state.muted,
      };
    }
    case types.ON_SEEK_MOUSE_DOWN: {
      return {
        ...state,
        seeking: true,
      };
    }
    case types.ON_SEEK_MOUSE_UP: {
      return {
        ...state,
        seeking: false,
      };
    }
    case types.ON_SEEK_CHANGE: {
      return {
        ...state,
        played: action.payload,
      };
    }
    case types.ON_VOLUME_CHANGE: {
      return {
        ...state,
        volume: action.payload,
      };
    }
    case types.SET_SEEK: {
      return {
        ...state,
        seek: action.payload,
      };
    }
    case types.CHANGE_SPEEDS_INDEX: {
      const idx =
        action.payload === "+" ? state.speedsIndex + 1 : state.speedsIndex - 1;
      return {
        ...state,
        speedsIndex: idx,
        playbackMultiplier: speeds[idx],
      };
    }
    // this.setState({ url: null, playing: false })
    default:
      // console.log("Failed Player action", action);
      return state;
  }
}
