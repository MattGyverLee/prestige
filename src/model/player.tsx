import "../App.css";

import * as actions from "../store";

import React, { Component } from "react";
import {
  faExpandArrowsAlt,
  faPause,
  faPlay
} from "@fortawesome/free-solid-svg-icons";

import Duration from "./duration";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { LooseObject } from "../store/annotations/types";
import ReactPlayer from "react-player";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import repeat from "../assets/icons/player/repeat.png";

// import fullscreen from "../assets/icons/player/fullscreen.png";
// import pause from "../assets/icons/player/pause.png";
// import play from "../assets/icons/player/play.png";

interface StateProps {
  controls?: boolean;
  duration: any;
  loop: boolean;
  muted: boolean;
  parent?: any;
  pip: boolean;
  playbackRate: number;
  played: number;
  playing: boolean;
  seek: number;
  seeking?: boolean;
  state?: any;
  url: string;
  volume: number;
  loaded: number;
  sourceMedia: LooseObject[];
  annotMedia: LooseObject[];
}

interface DispatchProps {
  playPause: typeof actions.playPause;
  stopPlaying: typeof actions.stopPlaying;
  toggleLoop: typeof actions.toggleLoop;
  onPlay: typeof actions.onPlay;
  onEnded: typeof actions.onEnded;
  onProgress: typeof actions.onProgress;
  play: typeof actions.play;
  setURL: typeof actions.setURL;
  setDuration: typeof actions.setDuration;
  setPlaybackRate: typeof actions.setPlaybackRate;
  setSeek: typeof actions.setSeek;
  toggleMuted: typeof actions.toggleMuted;
  onSeekMouseDown: typeof actions.onSeekMouseDown;
  onSeekMouseUp: typeof actions.onSeekMouseUp;
  onSeekChange: typeof actions.onSeekChange;
  onVolumeChange: typeof actions.onVolumeChange;
}

interface PlayerProps extends StateProps, DispatchProps {}

class PlayerZone extends Component<PlayerProps> {
  private player!: ReactPlayer;

  private speeds: number[] = [0.2, 0.33, 0.5, 0.66, 0.8, 1, 1.25, 1.5, 2, 3, 5];
  private speedsIndex: number = 5;

  componentDidUpdate() {
    if (this.props.seek !== -1) {
      this.player.seekTo(this.props.seek);
      this.props.setSeek(-1, -1);
    }
  }

  pip = () => {
    // this.setState({ pip: !this.props.pip });
  };

  onVolumeChange = (e: any) => {
    this.props.onVolumeChange(parseFloat(e.target.value));
  };

  minusPlaybackRate = () => {
    if (this.speedsIndex > 0) {
      this.speedsIndex--;
      this.props.setPlaybackRate(this.speeds[this.speedsIndex]);
    }
  };

  plusPlaybackRate = () => {
    if (this.speedsIndex < this.speeds.length - 1) {
      this.speedsIndex++;
      this.props.setPlaybackRate(this.speeds[this.speedsIndex]);
    }
  };

  onPause = () => {
    console.log("onPause");
  };

  onSeekMouseDown = (e: any) => {
    this.props.onSeekMouseDown();
  };

  onSeekChange = (e: any) => {
    this.props.onSeekChange(parseFloat(e.target.value));
  };

  onSeekMouseUp = (e: any) => {
    this.props.onSeekMouseUp();
    this.props.setSeek(
      parseFloat(e.target.value),
      parseFloat(e.target.value) * this.props.duration,
      0
    );
  };

  onPlay = () => {
    this.props.onPlay();
  };

  loadNewFile(blobURL: string) {
    this.props.play();
    this.props.setURL(blobURL);
  }

  onDuration = (duration: number) => {
    console.log("onDuration", duration);
    this.props.setDuration(duration);
  };

  onClickFullscreen = () => {
    // screenfull.request(findDOMNode(this.player))
  };

  ref = (player: any) => {
    this.player = player;
  };

  render() {
    return (
      <div>
        <div className="player-wrapper">
          <ReactPlayer
            className="react-player"
            height="70%"
            loop={this.props.loop}
            muted={this.props.muted}
            onBuffer={() => console.log("onBuffer")}
            onDuration={this.onDuration}
            onEnded={this.props.onEnded}
            onError={e => console.log("onError", e)}
            onPause={this.onPause}
            onPlay={this.onPlay}
            onProgress={this.props.onProgress}
            onReady={() => console.log("onReady")}
            onSeek={e => console.log("onSeek", e)}
            onStart={() => console.log("onStart")}
            pip={this.props.pip}
            playbackRate={this.props.playbackRate}
            playing={this.props.playing}
            ref={this.ref}
            url={this.props.url}
            volume={this.props.volume}
            width="100%"
          />
          <div className="control-row">
            <div className="control-row-items">
              <button
                className="play-pause-button"
                onClick={this.props.playPause}
              >
                <FontAwesomeIcon
                  icon={this.props.playing ? faPause : faPlay}
                  color="black"
                ></FontAwesomeIcon>
              </button>
              <button
                className="loop-button"
                id="loop"
                onClick={this.props.toggleLoop}
              >
                <img width="20px" src={repeat} alt="Loop Icon"></img>
              </button>
              <button onClick={this.minusPlaybackRate}>-</button>
              <div className="playback-rate">{this.props.playbackRate}x</div>
              <button onClick={this.plusPlaybackRate}>+</button>
              <input
                className="seek-input"
                max={1}
                min={0}
                onChange={this.onSeekChange}
                onMouseDown={this.onSeekMouseDown}
                onMouseUp={this.onSeekMouseUp}
                step="any"
                type="range"
                value={this.props.played}
              />
              <div className="durations">
                <Duration
                  className="duration-elapsed"
                  seconds={this.props.duration * this.props.played}
                />
                <Duration
                  className="total-duration"
                  seconds={this.props.duration}
                />
              </div>
              <button
                className="fullscreen-button"
                onClick={this.onClickFullscreen}
              >
                <FontAwesomeIcon
                  icon={faExpandArrowsAlt}
                  color="black"
                ></FontAwesomeIcon>
              </button>
            </div>
          </div>
          <div className="current-transcription"></div>
        </div>
        <div className="audio-Layers">
          <div>
            <input
              id="contactChoice1"
              name="contact"
              type="radio"
              value="email"
            />
            <label htmlFor="contactChoice1">Translation</label>
            <input
              id="contactChoice2"
              name="contact"
              type="radio"
              value="phone"
            />
            <label htmlFor="contactChoice2">Original</label>
            <input
              id="contactChoice3"
              name="contact"
              type="radio"
              value="mail"
            />
            <label htmlFor="contactChoice3">Careful</label>
          </div>
        </div>
        <div className="mediaControls">
          <table>
            <tbody>
              <tr>
                <th>Volume</th>
                <td>
                  <input
                    max={1}
                    min={0}
                    onChange={this.onVolumeChange}
                    step="any"
                    type="range"
                    value={this.props.volume}
                  />
                </td>
              </tr>
              <tr>
                <th>
                  <label htmlFor="muted">Muted</label>
                </th>
                <td>
                  <input
                    checked={this.props.muted}
                    id="muted"
                    onChange={this.props.toggleMuted}
                    type="checkbox"
                  />
                </td>
              </tr>
              <tr>
                <th>remaining</th>
                <td>
                  <Duration
                    className="Duration-Remaining"
                    seconds={(
                      this.props.duration *
                      (1 - this.props.played)
                    ).toFixed(3)}
                  />
                </td>
              </tr>
              <tr>
                <th>Loaded</th>
                <td>
                  <progress max={1} value={this.props.loaded} />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state: actions.StateProps): StateProps => ({
  playbackRate: state.player.playbackRate,
  played: state.player.played,
  muted: state.player.muted,
  controls: state.player.controls,
  duration: state.player.duration,
  playing: state.player.playing,
  pip: state.player.pip,
  seek: state.player.seek,
  url: state.player.url,
  loop: state.player.loop,
  volume: state.player.volume,
  loaded: state.player.loaded,
  sourceMedia: state.tree.sourceMedia,
  annotMedia: state.tree.annotMedia
});

const mapDispatchToProps = (dispatch: any): DispatchProps => ({
  ...bindActionCreators(
    {
      play: actions.play,
      setURL: actions.setURL,
      playPause: actions.playPause,
      stopPlaying: actions.stopPlaying,
      toggleLoop: actions.toggleLoop,
      onPlay: actions.onPlay,
      onEnded: actions.onEnded,
      onProgress: actions.onProgress,
      setDuration: actions.setDuration,
      setPlaybackRate: actions.setPlaybackRate,
      setSeek: actions.setSeek,
      toggleMuted: actions.toggleMuted,
      onSeekMouseDown: actions.onSeekMouseDown,
      onSeekMouseUp: actions.onSeekMouseUp,
      onSeekChange: actions.onSeekChange,
      onVolumeChange: actions.onVolumeChange
    },
    dispatch
  )
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PlayerZone);

/*
                <img
                  width="17px"
                  src={this.props.playing ? pause : play}
                  alt="Play/Pause Icon"
                ></img>

                <img width="18px" src={fullscreen} alt="Fullscreen Icon"></img>

*/
