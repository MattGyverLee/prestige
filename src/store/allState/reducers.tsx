import * as types from "./types";

import { annCleanStore } from "../annotations/reducers";
import { playerCleanStore } from "../player/reducers";
import { systemCleanStore } from "../system/reducers";
import { treeCleanStore } from "../tree/reducers";

var emptyOverState: types.OverState = {
  name: "Prestige",
  system: undefined,
  tree: undefined,
  player: undefined,
  annotation: undefined
};
// Using a dummy property so that the rest can be undefined.
const initialOverState: types.OverState = {
  name: "Prestige",
  system: { ...systemCleanStore },
  tree: { ...treeCleanStore },
  player: { ...playerCleanStore },
  annotation: { ...annCleanStore }
};

export const rootReducer = (
  state: types.OverState = initialOverState,
  action: types.OverActionTypes
) => {
  switch (action.type) {
    case types.HARD_RESET_APP: {
      state = emptyOverState;
      return {
        emptyOverState
      };
    }
    case types.ON_NEW_FOLDER: {
      let tempState = initialOverState;
      return {
        ...tempState,
        tree: { ...tempState.tree, folderPath: action.payload }
      };
    }
    default:
      return state;
  }
};
