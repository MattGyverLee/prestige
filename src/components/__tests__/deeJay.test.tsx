// This comment Allows me to put Jest Calls before Imports

import "@testing-library/jest-dom/extend-expect";

import ConnectedDeeJay, { DeeJay } from "../DeeJay/DeeJay";

import { Provider } from "react-redux";
import React from "react";
import { annCleanStore } from "../../store/annot/reducers";
import { cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { deeJayCleanStore } from "../../store/deeJay/reducers";
import { playerCleanStore } from "../../store/player/reducers";
import { systemCleanStore } from "../../store/system/reducers";
import { treeCleanStore } from "../../store/tree/reducers";
import TestRenderer from "react-test-renderer";

const renderer = TestRenderer;

// jest.mock("wavesurfer.js");
jest.autoMockOn();

const store = configureMockStore([])({
  annot: { ...annCleanStore },
  deeJay: {
    ...deeJayCleanStore,
  },
  player: { ...playerCleanStore },
  system: { ...systemCleanStore },
  tree: { ...treeCleanStore },
});

let deejayMaster: any;

// TODO: Constrain to Media Type

afterEach(cleanup);

const props = {
  setWSVolume: jest.fn(),
  setWSDuration: jest.fn(),
  setSeek: jest.fn(),
  resetDeeJay: jest.fn(),
  setDispatch: jest.fn(),
  playerPlay: jest.fn(),
  waveformAdded: jest.fn(),
  volumes: [1, 0, 0],
};

afterEach(cleanup);

// it("is true", () => true);

it("renders deeJay without crashing", () => {
  renderer.act(() => {
    deejayMaster = renderer.create(
      <Provider store={store}>
        <ConnectedDeeJay {...props} />
      </Provider>
    );
  });
  expect(deejayMaster).toMatchSnapshot();
  const deejayHandle: DeeJay = deejayMaster.root.findByType(DeeJay).instance;
  expect(deejayHandle).toBeVisible();
});
