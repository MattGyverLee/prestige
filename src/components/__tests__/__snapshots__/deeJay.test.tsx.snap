// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders deeJay without crashing 1`] = `
<div>
  <div
    className="wave-table-container"
  >
    <table
      className="wave-table"
    >
      <tbody>
        <tr>
          <td
            className="wave-table-enable"
          >
            <div
              className="buttonWrapper"
            >
              <div
                className="ThreeDimButton"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <img
                  alt=""
                  className="black"
                  height={50}
                  src="disabled50.png"
                  width={50}
                />
                <div
                  className="overlay"
                >
                  <img
                    alt=""
                    className="green"
                    height={50}
                    src="enabled50.png"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    width={50}
                  />
                </div>
              </div>
            </div>
            High
          </td>
          <td
            className="wave-table-volume"
          >
            <input
              id="0"
              max={1}
              min={0}
              onChange={[Function]}
              step={0.01}
              type="range"
              value={1}
            />
          </td>
          <td
            className="waveform"
            id="waveform0"
          />
        </tr>
        <tr>
          <td
            className="wave-table-enable"
          >
            <div
              className="buttonWrapper"
            >
              <div
                className="ThreeDimButton"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <img
                  alt=""
                  className="black"
                  height={50}
                  src="disabled50.png"
                  width={50}
                />
                <div
                  className="overlay"
                >
                  <img
                    alt=""
                    className="green"
                    height={50}
                    src="enabled50.png"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    width={50}
                  />
                </div>
              </div>
            </div>
            Muted
          </td>
          <td
            className="wave-table-volume"
          >
            <input
              id="1"
              max={1}
              min={0}
              onChange={[Function]}
              step={0.01}
              type="range"
              value={0}
            />
          </td>
          <td
            className="waveform"
            id="waveform1"
          />
        </tr>
        <tr>
          <td
            className="wave-table-enable"
          >
            <div
              className="buttonWrapper"
            >
              <div
                className="ThreeDimButton"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <img
                  alt=""
                  className="black"
                  height={50}
                  src="disabled50.png"
                  width={50}
                />
                <div
                  className="overlay"
                >
                  <img
                    alt=""
                    className="green"
                    height={50}
                    src="enabled50.png"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    width={50}
                  />
                </div>
              </div>
            </div>
            Muted
          </td>
          <td
            className="wave-table-volume"
          >
            <input
              id="2"
              max={1}
              min={0}
              onChange={[Function]}
              step={0.01}
              type="range"
              value={0}
            />
          </td>
          <td
            className="waveform"
            id="waveform2"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
